**Linux** [Build 7877](http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=68012&buildTypeId=WFPR) outcome was **FAILURE** using a merge of 76cf4efb201987bb29d6f4f2c9606c8cb7fd4219
Summary: Tests failed: 1 (1 new), passed: 3027, ignored: 358 Build time: 00:55:35
### Failed tests
```
org.jboss.as.testsuite.integration.secman.PBStaticMethodsTestCase: 	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:83)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:64)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deploy(ArchiveDeployer.java:46)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.deploy(CommonDeployableContainer.java:144)
Caused by: java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.xnio.nio.WorkerThread$ConnectHandle.handleReady(WorkerThread.java:319)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:545)
	at ...asynchronous invocation...(Unknown Source)
	at org.jboss.remoting3.EndpointImpl.doConnect(EndpointImpl.java:272)
	at org.jboss.remoting3.EndpointImpl.doConnect(EndpointImpl.java:253)
	at org.jboss.remoting3.EndpointImpl.connect(EndpointImpl.java:351)
	at org.jboss.remoting3.EndpointImpl.connect(EndpointImpl.java:339)
	at org.jboss.as.protocol.ProtocolConnectionUtils.connect(ProtocolConnectionUtils.java:83)
	at org.jboss.as.protocol.ProtocolConnectionUtils.connectSync(ProtocolConnectionUtils.java:114)
	at org.jboss.as.protocol.ProtocolConnectionManager$EstablishingConnection.connect(ProtocolConnectionManager.java:257)
	at org.jboss.as.protocol.ProtocolConnectionManager.connect(ProtocolConnectionManager.java:71)
	at org.jboss.as.protocol.mgmt.FutureManagementChannel$Establishing.getChannel(FutureManagementChannel.java:212)
	at org.jboss.as.controller.client.impl.RemotingModelControllerClient.getOrCreateChannel(RemotingModelControllerClient.java:146)
	at org.jboss.as.controller.client.impl.RemotingModelControllerClient$1.getChannel(RemotingModelControllerClient.java:65)
	at org.jboss.as.protocol.mgmt.ManagementChannelHandler.executeRequest(ManagementChannelHandler.java:147)
	at org.jboss.as.protocol.mgmt.ManagementChannelHandler.executeRequest(ManagementChannelHandler.java:122)
	at org.jboss.as.controller.client.impl.AbstractModelControllerClient.executeRequest(AbstractModelControllerClient.java:263)
	at org.jboss.as.controller.client.impl.AbstractModelControllerClient.execute(AbstractModelControllerClient.java:168)
	at org.jboss.as.controller.client.impl.AbstractModelControllerClient.executeAsync(AbstractModelControllerClient.java:111)
	at org.jboss.as.controller.client.helpers.standalone.impl.ModelControllerClientServerDeploymentManager.executeOperation(ModelControllerClientServerDeploymentManager.java:50)
	at org.jboss.as.controller.client.helpers.standalone.impl.AbstractServerDeploymentManager.execute(AbstractServerDeploymentManager.java:79)
	at org.jboss.as.controller.client.helpers.standalone.ServerDeploymentHelper.deploy(ServerDeploymentHelper.java:54)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:77)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:64)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deploy(ArchiveDeployer.java:46)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.deploy(CommonDeployableContainer.java:144)
------- Stdout: -------
&amp#27;[0m14:06:08,694 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 5) WFLYUT0022: Unregistered web context: /deployment-no-perm
&amp#27;[0m&amp#27;[0m14:06:08,701 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-1) WFLYSRV0028: Stopped deployment deployment-no-perm.war (runtime-name: deployment-no-perm.war) in 8ms
&amp#27;[0m&amp#27;[0m14:06:08,711 INFO  [org.jboss.as.repository] (management-handler-thread - 2) WFLYDR0002: Content removed from location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/secman/target/jbossas/standalone/data/content/b6/9f2a8fd8e9bbfb40e0f9e3104ccd8f9d5f1294/content
&amp#27;[0m&amp#27;[0m14:06:08,712 INFO  [org.jboss.as.server] (management-handler-thread - 2) WFLYSRV0009: Undeployed "deployment-no-perm.war" (runtime-name: "deployment-no-perm.war")
&amp#27;[0m14:06:08,731 ERROR [org.jboss.as.cli.CommandContext] (main) {
    "outcome" => "failed",
    "failure-description" => "WFLYCTL0216: Management resource '[
    (\"subsystem\" => \"security-manager\"),
    (\"deployment-permissions\" => \"default\")
]' not found",
    "rolled-back" => true
}
14:06:08,746 ERROR [org.jboss.as.cli.CommandContext] (main) {"outcome" => "success"}
&amp#27;[0m14:06:08,780 INFO  [org.jboss.as.connector.subsystems.datasources] (MSC service thread 1-3) WFLYJCA0010: Unbound data source [java:jboss/datasources/ExampleDS]
&amp#27;[0m&amp#27;[0m14:06:08,781 INFO  [org.wildfly.extension.messaging-activemq] (MSC service thread 1-1) WFLYMSGAMQ0006: Unbound messaging object to jndi name java:jboss/DefaultJMSConnectionFactory
&amp#27;[0m&amp#27;[0m14:06:08,781 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-1) WFLYUT0019: Host default-host stopping
&amp#27;[0m&amp#27;[0m14:06:08,783 INFO  [org.jboss.as.connector.deployment] (MSC service thread 1-2) WFLYJCA0011: Unbound JCA ConnectionFactory [java:/JmsXA]


```

--- a/clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/logging/JGroupsLogger.java
+++ b/clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/logging/JGroupsLogger.java
@@ -126,9 +126,6 @@ public interface JGroupsLogger extends BasicLogger {
     @Message(id = 18, value = "Instantiation exception on converter for attribute/method %s")
     String instantiationExceptionOnConverterForAttribute(String attrName);
 
-    @Message(id = 19, value = "Protocol %s not found in current stack")
-    String protocolNotFoundInStack(String protocolName);
-
     @Message(id = 20, value = "Unable to load protocol class %s")
     String unableToLoadProtocol(String protocolName);
 

--- a/clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/ProtocolMetricsHandler.java
+++ b/clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/ProtocolMetricsHandler.java
@@ -263,8 +263,6 @@ public class ProtocolMetricsHandler extends AbstractRuntimeOnlyHandler {
                 } else {
                     context.getFailureDescription().set(JGroupsLogger.ROOT_LOGGER.unknownMetric(name));
                 }
-            } else {
-                context.getFailureDescription().set(JGroupsLogger.ROOT_LOGGER.protocolNotFoundInStack(protocolName));
             }
         } catch (ClassNotFoundException | ModuleLoadException e) {
             context.getFailureDescription().set(e.getLocalizedMessage());

--- a/testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/model/ReadFullModelTestCase.java
+++ b/testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/model/ReadFullModelTestCase.java
@@ -1,0 +1,89 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015, Red Hat Middleware LLC, and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.test.manualmode.model;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.INCLUDE_RUNTIME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.READ_RESOURCE_OPERATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RECURSIVE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+
+import org.jboss.arquillian.container.test.api.ContainerController;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.model.test.ModelTestUtils;
+import org.jboss.as.test.shared.ServerReload;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.dmr.ModelNode;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Ensures that the full model including runtime resources/attributes can be read in both normal and admin-only mode
+ * for the fullest server config possible
+ *
+ * @author Kabir Khan
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class ReadFullModelTestCase {
+
+    //This is the full-ha setup which is the fullest config we have
+    //Reuse this rather than a setup
+    public static final String SERVER = "jbossas-messaging-ha-server1";
+
+    @ArquillianResource
+    protected static ContainerController controller;
+
+    protected static ModelControllerClient createClient1() {
+        return TestSuiteEnvironment.getModelControllerClient();
+    }
+
+
+    @Test
+    public void test() throws Exception {
+        controller.start(SERVER);
+        try {
+            ModelControllerClient client = TestSuiteEnvironment.getModelControllerClient();
+            ModelNode rr = Util.createEmptyOperation(READ_RESOURCE_OPERATION, PathAddress.EMPTY_ADDRESS);
+            rr.get(INCLUDE_RUNTIME).set(true);
+            rr.get(RECURSIVE).set(true);
+            ModelNode result = ModelTestUtils.checkResultAndGetContents(client.execute(rr));
+
+            //Just a quick sanity test to check that we are full-ha by making sure some of the expected subsystems are there
+            Assert.assertTrue(result.hasDefined(SUBSYSTEM, "messaging-activemq"));
+            Assert.assertTrue(result.hasDefined(SUBSYSTEM, "jgroups"));
+
+            ServerReload.executeReloadAndWaitForCompletion(client, true);
+            result = ModelTestUtils.checkResultAndGetContents(client.execute(rr));
+        } finally {
+            if (controller.isStarted(SERVER)) {
+                controller.stop(SERVER);
+            }
+        }
+    }
+
+}

